backend/simulation/realm/graph.py

```diff
@@ -71,12 +77,18 @@ def addOutgoing(self, e: Edge) -> None:
     def from_json(json: Any) -> Node:
         if json["type"] == "junction":
             return Junction.from_json(json)
+        elif json["type"] == "depot":
+            return Depot.from_json(json)
         else:
             raise NotImplementedError
 
 class Junction(Node):
 
     def entry(self, truck: Truck) -> None:
+        super().entry(truck)
+        if truck.done():
+            return
+        self._trucks.pop()
         self._outgoing[self._routing_table[truck.destination]].entry(truck)
```

Comment:
is it a good idea to:
(a) let trucks finish in a junction
(b) store these trucks forever in the junction if they do finish there
?

Reply:
a) I think this should be checked in the configuration file, don't provide a configuration which terminates at a junction.
b) Probably best to delete trucks and just add a "done" id mapping in observation dictionary to save on memory usage. I don't think this is necessary for MVP.

Reply:
In reply to 4) delete lines 50-53 for most optimal behavior. I don't mind either way.

Reply:
Cool - my point was that we are currently allowing (a) and (b) to occur ( the highlighted code is within the `Junction` class)